### PR TITLE
Add support for assemblies with multiple mscorlib references

### LINKED ISSUE #512: An unexpected error was encountered while parsing the metadata for the PE file [System.InvalidOperationException: Sequence contains more than one matching element]
Hello,

I am facing the issue below when try to analyze one assembly.

```
C:\Sources\Repos\dotnet-apiport\bin\Debug\ApiPort\net46\win7-x64>ApiPort.exe analyze -f "f:\temp.dll" -t ".NET Framework, Version=4.6.2" -b -r Html -o "C:\temp.html"
Microsoft (R) API Portability Analyzer vunknown
Copyright (C) Microsoft Corporation. All rights reserved.

This tool analyzes .NET assemblies to determine possible problems moving between .NET platforms (such as
Windows Store, desktop, Mono, .NET Core, etc) as well as between .NET Framework versions (ie 4.x->4.y).

To learn more about how this tool works, including the data we are collecting,
go here - http://go.microsoft.com/fwlink/?LinkId=397652

Detecting assembly references                     [Failed]

**An unexpected error was encountered while parsing the metadata for the PE file located at f:\temp.dll**.

System.InvalidOperationException: Sequence contains more than one matching element
   at System.Linq.Enumerable.SingleOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
   at Microsoft.Fx.Portability.Analyzer.DependencyFinderEngineHelper.GetSystemRuntimeAssemblyInformation() in C:\Sources\Repos\dotnet-apiport\src\Microsoft.Fx.Portability.MetadataReader\DependencyFinderEngineHelper.cs:line 213
   at Microsoft.Fx.Portability.Analyzer.DependencyFinderEngineHelper.ComputeData() in C:\Sources\Repos\dotnet-apiport\src\Microsoft.Fx.Portability.MetadataReader\DependencyFinderEngineHelper.cs:line 60
   at Microsoft.Fx.Portability.Analyzer.ReflectionMetadataDependencyInfo.GetDependencies(IAssemblyFile file) in C:\Sources\Repos\dotnet-apiport\src\Microsoft.Fx.Portability.MetadataReader\ReflectionMetadataDependencyInfo.cs:line 128

```

I verified the source code and the issue is taking place in the line below

```
C:\Sources\Repos\dotnet-apiport\src\Microsoft.Fx.Portability.MetadataReader\DependencyFinderEngineHelper.cs:213
var matchingAssembly = microsoftAssemblies.SingleOrDefault(x => s_systemObjectAssemblies.Contains(x.Name));

```

microsoftAssemblies collection has duplicated assembly names that causes exception on SingleOrDefault() method as below:
```

[{System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{System.Configuration, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{System.Data, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{System.Data, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{System.Data.DataSetExtensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{System.Drawing, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{System.ServiceModel, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{System.Web, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{System.Web, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{System.Web.Extensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{System.Web.Services, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{mscorlib, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation
{mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089}	Microsoft.Fx.Portability.ObjectModel.AssemblyReferenceInformation](url)
```

### PR DESCRIPTION
**Pull request by twsouthwick**
Fixes #512

### CODE REVIEW

File: tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs [hunk 1]
```diff
@@ -17,7 +16,7 @@
 
 namespace Microsoft.Fx.Portability.MetadataReader.Tests
 {
-    public class ManagedMetadataReaderTests
+    public partial class ManagedMetadataReaderTests
```
Comment (conniey) [inline tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs]: Why the change to partial? I couldn't find another partial class matching this.

Comment (twsouthwick) [reply tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs]: Already fixed :) was auto-added during a refactor
